### ISSUE TITLE
Padrão de Código

### ISSUE THREAD

**Issue by HeavenVolkoff** [darthdevz/xplainshell-bot#1]
Sendo esse um projeto no qual todo mundo do grupo pode contribuir, acho interessante usarmos um padrão de código.

Pessoalmente eu uso o [StandardJS](https://standardjs.com/readme-ptbr.html).
Ele é muito facil de usar, só tem que fazer `npm install standard` e rodar `standard --fix` na pasta root do repositório pra ele corrigir a maior parte dos desvios cometidos. Entretanto pode ter gente que não goste das regras dele, ex: (exclusão do `;`).

Então seria uma boa listarmos aqui padrões que conheçamos para decidirmos qual seria o melhor de usar....

**Comment by victorhugorch** [comment]
Nunca usei mas sou totalmente a favor. (apesar de usar o ; no final.) 

Ele é similar ao [Eslint](http://eslint.org/)? Você já utilizou Vitor? @HeavenVolkoff

**Comment by HeavenVolkoff** [comment]
Sim, sempre uso nos meus projetos. O StandarJS usa o ESlint por baixo dos panos, ele é como se fosse um conjunto de regras predefinidas pro ESLint.

**Comment by victorhugorch** [comment]
Qual a maior vantagem em usar o StandarJS e não o ESLint direto? 

Vou sugerir a gente usar o ESLint porque acho que é uma experiência mais completa/pura pra quem nunca trabalhou com nenhuma ferramenta que avalia o seu padrão de código.

**Comment by fogodev** [comment]
@victorhugorch Webhook namoral agora

**Comment by luizfcneto** [comment]
Onde aperta pra curtir os comentários aqui? hahaha

**Comment by HeavenVolkoff** [comment]
Cara, pra mim a vantagem é não ter que configurar nada e seguir um padrão que uma quantidade razoável de gente também já está seguindo ou pelo menos conhece. A única vantagem que eu vejo do eslint é que a gente define o padrão e não se adapta à um. Quanto a quem nunca mexeu com esse tipo de ferramenta, eu não vejo muita diferença, o uso dos dois é o mesmo, no eslint vc só vai ter que configurá-lo antes, mas depois disso vc raramente vai mexer nele.

**Comment by victorhugorch** [comment]
@HeavenVolkoff Maneiro, gosto dessa parada de não ter que configurar nada. Não por não ter que configurar mas bem ou mal essa experiência de configurar o ESLint vai ser só pra quem implementar mesmo, não vejo muito ganho pra equipe inteira.

**Comment by graciano** [comment]
acho que configura logo tudo, mas se a pessoa não rodar os testes, ser tolerante ao erro.
Aliás, ser tolerante a qualquer tipo de erro, já que o objetivo do grupo é aprender mesmo ^^

**Comment by HeavenVolkoff** [comment]
@graciano eu não entendi o que vc quis dizer com `ser tolerante ao error`...

**Comment by graciano** [comment]
@HeavenVolkoff não encher o saco de quem não rodar o standardJS ao comitar ou cometer qualquer erro. Bom senso mesmo

**Comment by victorhugorch** [comment]
@graciano mas acho que faz parte do aprendizado a pessoa ser confrontada em rodar o StandardJS e se acostumar com essa prática, você não acha? Perde meio o sentido de adicionar um watcher do seu código se você pode cagar pra ele enquanto escreve código.

**Comment by graciano** [comment]
Alertar o erro não é ser intolerante. O que eu quis dizer é que realmente a decisão de colocar a parada não precisa ser impactada por alguém não saber usar. E que não tem problema as pessoas errarem pela natureza da ideia aqui.

**Comment by HeavenVolkoff** [comment]
@graciano O StandardJS não impede você de commitar nem nada..... vc te que rodar `standard` no terminal dentro do seu projeto se quiser verificar que erros existem, ou ter algum watcher ou integração com IDE que rode pra você. Eu pessoalmente tenho a integração com o Sublime, e antes de commitar sempre rodo `standard --fix` que tenta consertar ele mesmo os error que eu cometi contra o padrão.

**Comment by graciano** [comment]
to contemplado, usa tudo.
